### ISSUE TITLE
MySQL DELETE .. USING with JOIN produces duplicate table declarations in USING clause

### ISSUE THREAD

**Issue by wcurrie** [jOOQ/jOOQ#12194]
### Expected behavior

Mysql join example from https://github.com/jOOQ/jOOQ/issues/11925 should include one copy of `t` and `u`.

```java
ctx.deleteFrom(T)
        .using(T.join(U).on(T.I.eq(U.J)))
        .where(T.I.eq(1));
```

```sql
delete from `mcve`.`t`
using `mcve`.`t`
  join `mcve`.`u`
    on `mcve`.`t`.`i` = `mcve`.`u`.`j`
where `mcve`.`t`.`i` = 1
```

### Actual behavior

```sql
delete
from `mcve`.`t` using
     `mcve`.`t`, -- extra t here
     `mcve`.`t`
       join `mcve`.`u` on `mcve`.`t`.`i` = `mcve`.`u`.`j`
where `mcve`.`t`.`i` = ?
```

Fails with:
```
java.sql.SQLSyntaxErrorException: Not unique table/alias: 't'
```

### Steps to reproduce the problem

With https://github.com/jOOQ/jOOQ-mcve/pull/10 mysql accepts sql generated by 3.14.12 but not by 3.15.1.

### Versions

- jOOQ: 3.15.1
- Java: 11
- Database (include vendor): mysql
- OS: osx
- JDBC Driver (include name if inofficial driver): mysql

**Comment by lukaseder** [comment]
Thanks for your report and the reproducer. I can confirm this. There seems to be a regression introduced by the various fixes in this area for jOOQ 3.15.0. As a workaround, this should work:

```java
ctx.deleteFrom(T)
        .using(T, U) // Use a table list instead of a JOIN
        .where(T.I.eq(1))
        .and(T.I.eq(U.J))) // Move the JOIN predicate to the where clause
        .execute();
```

**Comment by wcurrie** [comment]
Perfect. Thanks for all your work.

**Comment by lukaseder** [comment]
Ah yes, we didn't have an integration test for this syntax, unfortunately, because it can't currently be emulated in PostgreSQL. That doesn't mean we shouldn't support it for MySQL, of course.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12204). Thanks again for your report!
